hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/handlers/NodeCreateHandler.java

```diff
@@ -0,0 +1,188 @@
+        validateFalsePreCheck(accountId.equals(AccountID.DEFAULT), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(!accountId.hasAccountNum() && accountId.hasAlias(), INVALID_NODE_ACCOUNT_ID);
+        validateFalsePreCheck(op.gossipEndpoint().isEmpty(), INVALID_GOSSIP_ENDPOINT);
+        validateFalsePreCheck(op.serviceEndpoint().isEmpty(), INVALID_SERVICE_ENDPOINT);
+        validateFalsePreCheck(op.gossipCaCertificate().length() == 0, INVALID_GOSSIP_CAE_CERTIFICATE);
+    }
+
+    @Override
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
+        requireNonNull(context);
+        final var op = context.body().nodeCreate();
+        final var accountStore = context.createStore(ReadableAccountStore.class);
+        final var accountId = op.accountIdOrElse(AccountID.DEFAULT);
+
+        validateTruePreCheck(accountStore.contains(accountId), INVALID_NODE_ACCOUNT_ID);
+    }
+
+    /**
+     * Given the appropriate context, creates a new topic.
+     *
+     * @param handleContext the {@link HandleContext} for the active transaction
+     * @throws NullPointerException if one of the arguments is {@code null}
+     */
+    @Override
+    public void handle(@NonNull final HandleContext handleContext) {
+        requireNonNull(handleContext);
+
+        final var op = handleContext.body().nodeCreate();
+
+        final var configuration = handleContext.configuration();
+        final var nodeConfig = configuration.getConfigData(NodesConfig.class);
+        final var nodeStore = handleContext.writableStore(WritableNodeStore.class);
+
+        final var nodeBuilder = new Node.Builder();
+
+        validateFalse(nodeStore.sizeOfState() >= nodeConfig.maxNumber(), MAX_NODE_HAVE_BEEN_CREATED);
+        validateDescription(op.description(), nodeConfig);
+        validateGossipEndpoint(op.gossipEndpoint(), nodeConfig);
+        validateServiceEndpoint(op.serviceEndpoint(), nodeConfig);
+        validateFalse(op.gossipCaCertificate().length() == 0, INVALID_GOSSIP_CAE_CERTIFICATE);
+
+        nodeBuilder
+                .accountId(op.accountId())
+                .description(op.description())
+                .gossipEndpoint(op.gossipEndpoint())
+                .serviceEndpoint(op.serviceEndpoint())
+                .gossipEndpoint(op.gossipEndpoint())
+                .serviceEndpoint(op.serviceEndpoint())
+                .gossipCaCertificate(op.gossipCaCertificate())
+                .grpcCertificateHash(op.grpcCertificateHash())
+                .build();
+        final var node = nodeBuilder.nodeId(handleContext.newEntityNum()).build();
+
+        nodeStore.put(node);
+
+        final var recordBuilder = handleContext.recordBuilder(NodeCreateRecordBuilder.class);
+
+        recordBuilder.nodeID(node.nodeId());
+    }
+
+    private void validateDescription(@Nullable final String description, @NonNull final NodesConfig nodesConfig) {
```

Comment:
I think this method can be in helped we can reuse it in update

Reply:
same above.

Reply:
Added validator.